tests/Functional/Action/Lti/UpdateLti1p3OutcomeActionTest.php

```diff
@@ -0,0 +1,230 @@
+        );
+
+        $response = $this->kernelBrowser->getResponse();
+        self::assertInstanceOf(Response::class, $response);
+        self::assertEquals(Response::HTTP_UNAUTHORIZED, $response->getStatusCode());
+        self::assertStringContainsString(
+            'A Token was not found in the TokenStorage',
+            (string)$response->getContent()
+        );
+    }
+
+    public function testItReturnsUnauthorizedResponseWithInvalidToken(): void
+    {
+        $this->kernelBrowser->request(
+            'POST',
+            '/api/v1/lti1p3/outcome',
+            [],
+            [],
+            [
+                'CONTENT_TYPE' => 'text/xml',
+                'HTTP_AUTHORIZATION' => 'Bearer invalid'
+            ],
+            $this->getValidReplaceResultRequestXml()
+        );
+
+        $response = $this->kernelBrowser->getResponse();
+        self::assertInstanceOf(Response::class, $response);
+        self::assertEquals(Response::HTTP_UNAUTHORIZED, $response->getStatusCode());
+        self::assertStringContainsString('The JWT string must have two dots', (string)$response->getContent());
+    }
+
+    public function testItReturns400IfTheAuthenticationWorksButTheXmlIsInvalid(): void
+    {
+        $accessToken = $this->createTestClientAccessToken(
+            $this->registration,
+            ['https://purl.imsglobal.org/spec/lti-bo/scope/basicoutcome']
+        );
+        $authorization = sprintf('Bearer %s', $accessToken);
+
+        $this->kernelBrowser->request(
+            'POST',
+            '/api/v1/lti1p3/outcome',
+            [],
+            [],
+            [
+                'CONTENT_TYPE' => 'text/xml',
+                'HTTP_AUTHORIZATION' => $authorization
+            ],
+            'invalidXml'
+        );
+
+        self::assertEquals(Response::HTTP_BAD_REQUEST, $this->kernelBrowser->getResponse()->getStatusCode());
+        self::assertEquals(Assignment::STATE_READY, $this->getAssignment()->getState());
+    }
+
+    public function testItReturns404IfTheAuthenticationWorksButTheAssignmentDoesNotExist(): void
+    {
+        $accessToken = $this->createTestClientAccessToken(
+            $this->registration,
+            ['https://purl.imsglobal.org/spec/lti-bo/scope/basicoutcome']
+        );
+        $authorization = sprintf('Bearer %s', $accessToken);
+
+        $this->kernelBrowser->request(
+            'POST',
+            '/api/v1/lti1p3/outcome',
+            [],
+            [],
+            [
+                'CONTENT_TYPE' => 'text/xml',
+                'HTTP_AUTHORIZATION' => $authorization
+            ],
+            $this->getValidReplaceResultRequestXmlWithWrongAssignment()
+        );
+
+        self::assertEquals(Response::HTTP_NOT_FOUND, $this->kernelBrowser->getResponse()->getStatusCode());
+        self::assertEquals(Assignment::STATE_READY, $this->getAssignment()->getState());
+    }
+
+    private function getAssignment(): Assignment
```

Comment:
suggestion: I'd rename this method to assertAssigmentState and add inside of it:
self::assertEquals(Assignment::STATE_READY, $this->getAssignment()->getState());

Reply:
Followed suggestion but moved to a trait since it is used on both endpoint tests